### PR TITLE
O3-1990: Allow string as value for markdown

### PR DESCRIPTION
**Pull request by reagan-meant**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

Here is the sample json markdown being used. The first part part shows the current implementation using array while the subsequent ones show changes in this PR with string as value and render to handle unsupported type
```json
{
  "questions": [
    {
      "type": "markdown",
      "questionOptions": {
        "rendering": "markdown"
      },
      "id": "fooMarkdown",
      "value": [
        "**This form is used to:** Document differentiated service deliver enrolment/ disenrolment.",
        "\n\ntest 1",
        "_test 3_"
      ]
    },
    {
      "type": "markdown",
      "questionOptions": {
        "rendering": "markdown"
      },
      "id": "fooMarkdown2",
      "value": "**This form is used to:** \n\n Document _differentiated_ service deliver enrolment/ disenrolment."
    },
    {
      "type": "markdown",
      "questionOptions": {
        "rendering": "markdown"
      },
      "id": "fooMarkdown3",
      "value": {
        "wrongvalue": "wrong value type"
      }
    },
    {
      "label": "Admission Date",
      "type": "encounterDatetime",
      "questionOptions": {
        "rendering": "ui-select-extended"
      },
      "id": "encDate"
    }
  ]
}
```

## Screenshots
Screenshot 2024-04-16 at 13 01 15

## Related Issue

https://openmrs.atlassian.net/browse/O3-1990

### CODE REVIEW

File: src/components/inputs/markdown/ohri-markdown.component.tsx [hunk 1]
```diff
@@ -2,7 +2,15 @@ import React from 'react';
...
+const OHRIMarkdown: React.FC<OHRIFormFieldProps> = ({ question }) =>{
+  if (!question.isHidden) {
+    if (typeof question.value === 'string') {
+      return <ReactMarkdown>{question.value}</ReactMarkdown>;
+    } else if (Array.isArray(question.value)) {
+      return <ReactMarkdown children={question.value.join('\n')} />;
+    } else {
+      return <div>Invalid value type to render</div>;
+    }
+  }
```
Comment (samuelmale) [inline src/components/inputs/markdown/ohri-markdown.component.tsx:14]: @reagan-meant I would simply:

```jsx
const markdownValue = Array.isArray(question.value) ? question.value.join('\n') : question.value;
return !question.isHidden && <ReactMarkdown>{markdownValue}</ReactMarkdown>;
```

Two more issues:

1. Did we agree upon supporting a limited number of elements?
2. You seem to have a formatting issue on line 5

**Comment by reagan-meant** [comment]
```
{
              "type": "markdown",
              "questionOptions": {
                "rendering": "markdown"
              },
              "id": "fooMarkdown2",
              "value": "**This form is used to:** \n\n Document _differentiated_ service deliver enrolment/ disenrolment."
            },
            {
              "type": "markdown",
              "questionOptions": {
                "rendering": "markdown"
              },
              "id": "fooMarkdown3",
              "value": [
                "**This form is used to:** ",
                " \n\n Document _differentiated_ service deliver enrolment/ disenrolment."
              ]
}

**Comment by samuelmale** [comment]
This is close to the finish line; I think we also need to update [this](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/ohri-form.component.tsx#L321-L323) @reagan-meant

**Comment by samuelmale** [comment]
@reagan-meant basically this is good to go but I'm worried about the code duplication. What do you think of us introducing wrapper component?

markdown-wrapper.component.tsx:

```tsx

const MarkdownWrapper: React.FC<{ markdown: string | string[] }> = ({ markdown }) => {
  return (
    <ReactMarkdown
      children={Array.isArray(markdown) ? markdown.join('\n') : markdown}
      unwrapDisallowed={true}
      allowedElements={['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p', 'strong', 'em']}
    />
  );
};

```

src/components/inputs/markdown/ohri-markdown.component.tsx:

```tsx
const OHRIMarkdown: React.FC<OHRIFormFieldProps> = ({ question }) =>
  !question.isHidden && <MarkdownWrapper markdown={question.value} />;

```

**Pull request merged**